core/utils.c

```diff
@@ -726,7 +724,7 @@ size_t utils_base64Encode(uint8_t * dataP,
 
     result_len = prv_getBase64Size(dataLen);
 
-    if (result_len > dataLen) return 0;
+    if (result_len < dataLen) return 0;
 
```

Comment:
Actually this should be a comparaison with dataLen to be sure the provided buffer is large enough. So the fix should be:
```if (result_len > bufferLen) return 0;```